lib/common/client/middleware/logging.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Common
+  module Client
+    module Middleware
+      class Logging < Faraday::Middleware
+        DEFAULT_TYPE_KEY = 'undefined'
+
+        def initialize(app, type_key)
+          super(app)
+          @type_key = type_key || DEFAULT_TYPE_KEY
+        end
+
+        def call(env)
+          request_body = Base64.encode64(env.body)
+
+          @app.call(env).on_complete do |response_env|
+            PersonalInformationLog.create({
```

Comment:
Felt backgrounding this was premature. Could potentially cause some performance issues with large request/response bodies, but for this call at least it shouldn't be a problem.

Gut feeling is that this would cause more problems sending over to sidekiq for queue and processing in redis than it would just writing directly to the database, anyway.